streams/src/main/java/org/apache/kafka/streams/kstream/KStream.java

```diff
@@ -721,13 +725,31 @@ void writeAsText(final String filePath,
      *                    be used
      * @param topic       the topic name
      * @return a {@code KStream} that contains the exact same (and potentially repartitioned) records as this {@code KStream}
+     * @deprecated use {@code through(String, Produced)}
      */
+    @Deprecated
     KStream<K, V> through(final Serde<K> keySerde,
                           final Serde<V> valSerde,
                           final StreamPartitioner<? super K, ? super V> partitioner,
                           final String topic);
 
     /**
+     * Materialize this stream to a topic and creates a new {@code KStream} from the topic using the
+     * {@link Produced} instance for configuration of the key {@link Serde}, value {@link Serde}, and {@link StreamPartitioner}
+     * The specified topic should be manually created before it is used (i.e., before the Kafka Streams application is
+     * started).
+     * <p>
+     * This is equivalent to calling {@link #to(String, Produced) #to(String, Produced)}
```

Comment:
Nit: should be `{@link #to(String, Produced) #to(someTopicName, Produced.with(keySerde, valSerde))}`